mcs/class/corlib/System.Runtime.Remoting.Proxies/RealProxy.cs

```diff
@@ -49,8 +49,45 @@ namespace System.Runtime.Remoting.Proxies
 	[StructLayout (LayoutKind.Sequential)]
 	internal class TransparentProxy {
 		public RealProxy _rp;
-		IntPtr _class;
+		unsafe Mono.RuntimeStructs.RemoteClass* _class;
```

Comment:
What about we model this as follow:

Mono.RuntimeStructs.* < straight mapping of native structs
Mono.RuntimeHandles.* < wrapper Handle variants that contain any methods we might need.